### PR TITLE
feat(perl): Add complete Perl SDK implementation

### PR DESCRIPTION
**Pull request by clemensv**
## Summary

This PR introduces a complete, production-ready Perl SDK for JSON-Structure validation, achieving full parity with the Go SDK.

### Core Modules (6 modules, ~4,800 lines of code)

- **JSON::Structure** - Main entry point with convenient validation functions
- **JSON::Structure::SchemaValidator** - Validates JSON-Structure schema files against the meta-schema
- **JSON::Structure::InstanceValidator** - Validates JSON instances against JSON-Structure schemas
- **JSON::Structure::JsonSourceLocator** - Provides source location tracking for error messages
- **JSON::Structure::Types** - Type definitions and utilities
- **JSON::Structure::ErrorCodes** - Standardized error codes matching other SDKs

### Validation Capabilities

- All JSON-Structure type keywords: `type`, `abstract`, `ref`, `alt`
- Nested definitions and circular reference detection
- All format validators (uri, uri-reference, uri-template, email, date, time, uuid, ipv4, ipv6, etc.)
- All constraint keywords (minLength, maxLength, pattern, minimum, maximum, enum, const, etc.)
- Contains/minContains/maxContains for array validation
- If/then/else conditional composition
- Root schema anchoring with ``

### Test Suite

- **142 tests** across 6 test files
- Comprehensive adversarial test coverage
- All shared test assets from `test-assets/` directory
- Format validation tests
- Edge case and error handling tests

### Multi-Platform CI Workflow

Tests run on:
- **Perl versions**: 5.20, 5.26, 5.32, 5.38
- **Platforms**: Ubuntu, macOS, Windows

Workflow includes:
- Unit tests with parallel execution
- Perl::Critic linting
- Test coverage reporting
- Distribution build verification

### CPAN Publishing

- Automated publishing to CPAN on version tags (`v*`)
- Uses `CPAN::Uploader` with PAUSE credentials
- Publishing documentation in `PUBLISHING.md`

## Dependencies

**Runtime:**
- `JSON::PP` (core) - JSON parsing
- `MIME::Base64` (core) - Binary encoding
- `Scalar::Util` (core) - Type utilities
- `Math::BigInt` (core) - Large number handling
- `Time::Local` (core) - Date/time validation
- `B` (core) - Perl internals for type detection

**Test:**
- `Test::More` - Testing framework
- `File::Find` - Test asset discovery
- `Data::Dumper` - Debugging output

## Compatibility

- Minimum Perl version: **5.20** (uses signatures feature)
- No non-core dependencies for runtime
- Pure Perl implementation (no XS)

## Files Changed

| Category | Files | Lines |
|----------|-------|-------|
| Modules | 6 | ~4,800 |
| Tests | 6 | ~3,200 |
| CI/CD | 1 | 241 |
| Config | 4 | ~160 |
| Docs | 2 | ~450 |

## Checklist

- [x] All 142 tests pass on Perl 5.20, 5.26, 5.32, 5.38
- [x] Tests pass on Ubuntu, macOS, Windows
- [x] Code follows Perl best practices (`use strict`, `use warnings`)
- [x] All modules have `\` for CPAN
- [x] Documentation complete (README.md, POD, PUBLISHING.md)
- [x] CI workflow with linting, coverage, and build verification
- [x] CPAN publishing workflow ready

### CODE REVIEW

File: perl/README.md [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+### From CPAN (when available)
+
+```bash
+cpanm JSON::Structure
+```
+
+### From Source
+
+```bash
+git clone https://github.com/json-structure/sdk.git
```
Comment (hashbangperl) [inline perl/README.md]: cpanm will install directly from github: https://gist.github.com/keisatou/7499464?permalink_comment_id=4193350#gistcomment-4193350

File: perl/README.md [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+            $error->code;
+    }
+}
+```
+
+### Instance Validation
+
+```perl
+use JSON::Structure::InstanceValidator;
+use JSON::PP;
```
Comment (hashbangperl) [inline perl/README.md]: JSON::MaybeXS

File: perl/cpanfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# JSON::Structure Perl SDK dependencies
+# Run: cpanm --installdeps .
+
+requires 'perl', '5.020';
+
+# Core dependencies (most are in core, but specifying versions for clarity)
+requires 'JSON::PP', '2.0';
+requires 'MIME::Base64', '3.0';
+requires 'Scalar::Util', '1.0';
+requires 'Exporter', '5.0';
```
Comment (hashbangperl) [inline perl/cpanfile]: Again this is core so doesn't need to listed as a dependency

File: perl/lib/JSON/Structure/InstanceValidator.pm [hunk 4]
```diff
@@ -0,0 +1,1756 @@
...
+}
+
+sub _check_type {
+    my ($self, $value, $type) = @_;
+    
+    if ($type eq 'null') {
+        return !defined $value || (ref($value) eq '' && $value eq 'null');
+    }
+    elsif ($type eq 'boolean') {
+        return JSON::PP::is_bool($value) ||
```
Comment (hashbangperl) [inline perl/lib/JSON/Structure/InstanceValidator.pm]: use Scalar::Util 'blessed';
blessed $x && $x ->isa($class); for applicable json boolean types

**Review by hashbangperl — COMMENTED** [review]
First quick look, there's a few low hanging fruit improvements to be made. The validation could probably use some nicer cpan modules than simple regexps.
